react-ui/src/components/LogInForm.js

```diff
@@ -34,73 +34,93 @@ class logInForm extends Component {
-    .then(res => {
-      if(res.data!==""){
-        this.setState({
-          email : res.data.email,
-          userType: res.data.userType,
-          firstName: res.data.firstName,
-          personId: res.data.workerId || res.data.studentId,
-          accessToken: res.data.accessToken,
-          islogged: true
+    let validinput = true;
+    var params = { email: email, password: password }
+
+    if (email === "" && password === "") {
+      validinput = false
+      alert("Please enter email and password")
+      event.preventDefault();
+    } else if (email === "") {
+      validinput = false;
+      alert("Please enter an email")
+      event.preventDefault();
+    } else if (password === "") {
+      validinput = false;
+      alert("please enter an password")
+      event.preventDefault();
+    }
+
+    if (validinput === true) {
+      axios.get(`/public/login/?${queryString.stringify(params)}`)
+        .then(res => {
+          if (res.data !== "") {
+            this.setState({
+              email: res.data.email,
+              userType: res.data.userType,
+              firstName: res.data.firstName,
+              personId: res.data.workerId || res.data.studentId,
+              accessToken: res.data.accessToken,
+              islogged: true
+            })
+            // localStorage.setItem("token", "T");
+            console.log(this.state.userType)
+          } else {
+            alert("Invalid Email or Password")
+            event.target.reset();
+          }
         })
-        // localStorage.setItem("token", "T");
-        console.log(this.state.userType)
-      }
-    })
-    event.preventDefault();
-  } 
+      event.preventDefault();
+    }
+  }
 
   render() {
     // TO DO: Find a better way to pass the params
-    let newRoute= <Route exact path="/login" render={props => ( <Redirect to={`/dashboard/${this.state.loginParams.email}/${this.state.userType}/${this.state.firstName}/${this.state.personId}/${this.state.accessToken}`} Component={Dashboard}/>)}></Route> 
-  
+    let newRoute = <Route exact path="/login" render={props => (<Redirect to={`/dashboard/${this.state.loginParams.email}/${this.state.userType}/${this.state.firstName}/${this.state.personId}/${this.state.accessToken}`} Component={Dashboard} />)}></Route>
+
     if (this.state.islogged) {
       return newRoute;
     }
-      return (
-        <Container className="Form-container">
-           <Title name= "Log In."></Title>
-          <Row>
-           <Col sm={12} align="center">
-           <form onSubmit={this.login}>
-                <label>
-                  <input 
-                  className ="InputFields" 
-                  type="text" 
+    return (
+      <Container className="Form-container">
+        <Title name="Log In."></Title>
+        <Row>
+          <Col sm={12} align="center">
+            <form onSubmit={this.login}>
+              <label>
+                <input
+                  className="InputFields"
```

Comment:
Hi Kyle,
If you add the `required` keyword here, the above logic for rendering the alert boxes based on the state values for email and password would be unneeded. Thus, to simplify and reduce the redundancy in the code, I recommend leveraging the `required` keyword for these mandatory inputs.

Reply:
done